### PR TITLE
Fixed wrong keys used to WATCH dependencies

### PR DESCRIPTION
**Pull request by th3hamm0r**
**TL;DR:**
This bug has opened a lot of possible race-conditions, since the watch-logic from redis did not fail anymore, if dependencies have been changed in parallel.

This seems related to #758, especially https://github.com/rq/rq/issues/758#issuecomment-791603930, where jobs were hanging in deferred state, but I can't say, that my changes fix all issues in #758.

**Details:**
I found this issue during an upgrade of a project from RQ 0.13.0 to 1.10.1, which heavily uses job dependencies. Fortunately, during first tests I saw hanging jobs in "deferred" state.
Actually, I'm a co-author of the whole watch-logic for dependencies introduced in RQ 0.7.1, which has fixed race-conditions for single dependencies here: https://github.com/rq/rq/issues/739
Unfortunately, with the switch to multiple job dependencies (in RQ 1.2), the job-ids of dependencies have not been prefixed correctly, so probably since 1.2 the whole watch-logic basically does not work correctly, which results in race-conditions.

### CODE REVIEW

**Comment by th3hamm0r** [comment]
Fyi: creating tests for these race-conditions is really hard. I've tried to modify the original test from https://github.com/rq/rq/pull/742/files#diff-5c94530959bc0ef10f7a2c0fe76bd4145e10bf6c377627c10042dab0ebcf5c4aR572, but failed... I would still leave that original test, even when it did not "find" the current issue, but it tests some race-issues from the past.

**Comment by selwin** [comment]
Thanks!
